src/cmd/hostnames.ts

```diff
@@ -39,3 +54,23 @@ hostnames
         throw e
     }
   })
+
+function getErrorMessages(res: any): string[] {
+  if (res.data.errors) {
+    return res.data.errors.map((err: any) => errorMessage(err))
+  }
+  return []
+}
+
+function errorMessage(err: any): string {
+  if (typeof err === 'string') {
+    return err
+  } else if (err.title && err.detail) {
+    return `${err.title}: ${err.detail}`
+  } else if (err.title) {
+    return err.title
+  } else if (err.detail) {
+    return err.detail
+  }
+  return ''
+}
```

Comment:
Error handling looks generic enough that it could be handled at the API client level. Although I'm not sure what kind of return value you'd be sending back to the caller.

Maybe it's generic util to handle JSONAPI-style errors :)

Reply:
That's what I was thinking. `API.errorMessages` seemed somewhat awful, especially that the curent API is nice and small. I will try to ponder that some more.